### PR TITLE
Fix remove labels

### PR DESCRIPTION
**Pull request by nodece**
### Motivation

If different labels are selected multiple times, the previously selected labels cannot be deleted correctly.

Tests have been added and tested in my repo, see the following log:

```
[INFO] 2022/09/08 09:09:44 @Start docbot
[INFO] 2022/09/08 09:09:44 @EventName is PR
[INFO] 2022/09/08 09:09:44 PR description: Signed-off-by: Zixuan Liu <nodeces@gmail.com>

test


- [ ] `doc-not-needed`
- [x] `doc` 

[INFO] 2022/09/08 09:09:44 @List repo labels
[INFO] 2022/09/08 09:09:44 Repo labels: map[bug:{} doc:{} doc-complete:{} doc-label-missing:{} doc-not-needed:{} doc-required:{} documentation:{} duplicate:{} enhancement:{} good first issue:{} help wanted:{} invalid:{} question:{} wontfix:{}]
[INFO] 2022/09/08 09:09:44 PR labels: map[doc-not-needed:{} duplicate:{} enhancement:{}]
[INFO] 2022/09/08 09:09:44 Expected labels: map[doc:true doc-not-needed:false]
[INFO] 2022/09/08 09:09:44 Labels to add: [doc]
[INFO] 2022/09/08 09:09:45 Labels to remove: [doc-not-needed]
```

/cc @maxsxu @tisonkun @Anonymitaet

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@maxsxu 
could you please review this PR from a technical perspective? Thank you!

**Comment by tisonkun** [comment]
Can I summarize the issue on the current master is that the `doc` related labels can be more than they should be?

That's, no comments mangled regression as https://github.com/apache/pulsar-test-infra/issues/47, or the box is checked but still complain doc-label-missing.

**Comment by tisonkun** [comment]
I think we should try to decide which input is the source. Right now,

1. A change on labels manually by committers can trigger PR description modified.
2. A change in PR description can update labels.

I suppose we just use the PR description checkbox as the source, and this can also resolve @michaeljmarshall's #47 that the bot should not update PR description.

File: docbot/action.go [hunk 1]
```diff
@@ -47,11 +48,17 @@ func NewActionWithClient(ctx context.Context, ac *ActionConfig, client *github.C
 	}
 }
 
+const openedActionType = "opened"
+const editedActionType = "edited"
+const unlabeledActionType = "unlabeled"
+const labeledActionType = "labeled"
```
Comment (maxsxu) [inline docbot/action.go:54]: Suggest using one const group like:
```

const (
	openedActionType = "opened"
	editedActionType = "edited"
	unlabeledActionType = "unlabeled"
	labeledActionType = "labeled"
)
```

**Comment by nodece** [comment]
We have discussed this offline, using PR description as the only input, thanks all.

File: docbot/action.go [hunk 2]
```diff
@@ -47,11 +47,16 @@ func NewActionWithClient(ctx context.Context, ac *ActionConfig, client *github.C
...
+	openedActionType = "opened"
+	editedActionType = "edited"
+)
+
 func (a *Action) Run(prNumber int, actionType string) error {
 	a.prNumber = prNumber
 
 	switch actionType {
-	case "opened", "edited", "labeled", "unlabeled":
+	case openedActionType, editedActionType:
```
Comment (tisonkun) [inline docbot/action.go:59]: I think we should handle labeled/unlabeled action here to undo unintended editions?

For example, if only "doc-not-needed" box is checked and a committer check "doc" label, we should remove it.

**Review by tisonkun — APPROVED** [review]
LGTM. I think this can be one reason where Docker action is better, lol - then we can use @nodece's fork to test ourselves without changing the code :)

**Comment by nodece** [comment]
To publish to the Apache repository, we need PMC operations.

File: docbot/action.go [hunk 3]
```diff
@@ -130,10 +137,13 @@ func (a *Action) checkLabels() error {
...
-			}
-			if checked {
-				labelsToAdd[label] = struct{}{}
+				if !checked {
+					labelsToRemove[label] = struct{}{}
+				}
+			} else {
+				if checked {
+					labelsToAdd[label] = struct{}{}
+				}
```
Comment (maxsxu) [inline docbot/action.go:146]: Suggest making code more compact:
```
if _, found := prLabels[label]; found && !checked {
	labelsToRemove[label] = struct{}{}
} else if !found && checked {
	labelsToAdd[label] = struct{}{}
}
```

**Review by maxsxu — COMMENTED** [review]
Overall LGTM. left some comments

**Comment by tisonkun** [comment]
I noticed that this PR doesn't have a diff to remove PR edit logic. Will take a closer look now.

**Comment by nodece** [comment]
`addAndCleanupHelpComment()` method is safe, which doesn't add comments more than once.

**Comment by tisonkun** [comment]
@nodece since the manner changed, I think we should update the comment content also for simply inlining the information to ask the user to add a valid checkbox, instead of linking to the gdoc.

@nicoloboschi for your information. I noticed that you often omit the doc checkbox section and can be affected by what's achieved here.

**Pull request merged**